concepts/modularization/README.md

```diff
@@ -0,0 +1,75 @@
+# Modularization
+
+Kyma provides Kubernetes building blocks. It should be easy to pick only those that are needed for the job and it should be easy to add new blocks to extend Kyma features. With the growing number of components, it is not possible to always install them all anymore. 
+
+- [Independent releases of modules](#independent-releases-of-modules)
+- [Dependencies between components](#dependencies-between-components)
+- [Release channels](#release-channels)
+- [Versioning](#versioning)
+- [Manifest operator](#manifest-operator)
+
+# Independent releases of modules
+With the growing number of components, it is hard to deliver features and fixes quickly and efficiently. Changes in manifests require a new release of Kyma. Operators (reconcilers) are tightly coupled and must be released together. In most cases, new component releases don't involve any API changes and could be delivered in a few minutes. 
+
+
+# Dependencies between components
+Components can depend only on core Kubernetes API or API extensions introduced by other components. Component operators are responsible for checking if the required APIs are available and react properly to the missing dependencies by reducing functionality or even reporting errors. Component owners are responsible for integration tests with their dependencies (with all versions supported in official release channels).
+
+# Release channels
+Release channels let customers balance between available features and stability. The number of channels and their characteristics can be established later. Usually, projects introduce between 2 and 4 channels. Some examples:
+- [GKE](https://cloud.google.com/kubernetes-engine/docs/concepts/release-channels)
+- [Nextcloud](https://nextcloud.com/release-channels/)
+
+TO DO:
+- how release channels are implemented (one config file or folder in some repository / config maps)
+- what component versions are included in the channel 
+- governance model - requirements for promoting version to the stable channel (hotfix should be possible anyway)
+
+# Versioning
```

Comment:
Thanks - I will update the description to add more examples. I didn't include component sources in the versioning description because they are indirectly referenced by the component images. It doesn't matter if it is kyma component (like api-gateway) or 3rd party component (like prometheus). We need just a container image reference in the component deployment (chart or something else that operator can apply). Let me extend the versioning with some examples to be more clear.